core/components/blocks/Microcart/Product.js

```diff
@@ -1,53 +1,44 @@
-import { productThumbnailPath } from '@vue-storefront/store/helpers'
+import Product from '@vue-storefront/core/modules/cart/components/Product.ts'
 
 export default {
-  name: 'Product',
   data () {
+    // depreciated
     return {
       qty: 0,
       isEditing: false
     }
   },
-  props: {
-    product: {
-      type: Object,
-      required: true
-    }
-  },
-  computed: {
-    thumbnail () {
-      const thumbnail = productThumbnailPath(this.product)
-      if (typeof navigator !== 'undefined' && !navigator.onLine) {
-        return this.getThumbnail(thumbnail, 310, 300) // for offline support we do need to have ProductTile version
-      } else return this.getThumbnail(thumbnail, 150, 150)
-    }
+  beforeMount () {
+    // deprecated, will be moved to theme or removed in the near future #1742
+    this.$bus.$on('cart-after-itemchanged', this.onProductChanged)
   },
   beforeDestroy () {
+    // deprecated, will be moved to theme or removed in the near future #1742
     this.$bus.$off('cart-after-itemchanged', this.onProductChanged)
   },
-  beforeMount () {
-    this.$bus.$on('cart-after-itemchanged', this.onProductChanged)
-  },
   methods: {
-    removeItem () {
-      this.$store.dispatch('cart/removeItem', this.product)
-    },
-    onProductChanged (event) {
-      if (event.item.sku === this.product.sku) {
-        this.$forceUpdate()
-      }
-    },
     updateQuantity () {
+      // additional logic will be moved to theme
       this.qty = parseInt(this.qty)
       if (this.qty <= 0) {
         this.qty = this.product.qty
       }
-      this.$store.dispatch('cart/updateQuantity', { product: this.product, qty: this.qty })
+      Product.methods.updateQuantity.call(this, this.qty)
```

Comment:
Don't like that at all

Reply:
Why? It's kind of hack but its just for backward compatibility..It’s a solution to fact that mixed metods are overrided by methofs with the same names so if I want to extend it without repeating the code it's probably the only way